### DISCUSSION TITLE
Command history in Allure Report

### DISCUSSION THREAD

**Discussion by vitaly-kiselev-qa** [gemini-testing/testplane#924] [Q&A] [answered]
We use the plugin https://github.com/allure-framework/allure-js/tree/main/packages/hermione-allure to generate Allure Report.
It requiers either a `WebdriverIO.Browser` or `Mocha.Test` object to interact with the report.

We want to put the command history to the report:
https://github.com/gemini-testing/testplane/blob/master/docs/config.md#savehistorymode
The documentation states that the history is available only in `TEST_END`, `TEST_PASS` or `TEST_FAIL` event using the `TestResult` object. 
Looks like this object is not suiteable to interact with the Allure report.

Maybe, there is another way to do it?

**Comment by DudaGod** [comment, marked as answer]
Hello. Sorry for the long answer, I didn't notice notification in the mail about this discussion.

> We use the plugin https://github.com/allure-framework/allure-js/tree/main/packages/hermione-allure to generate Allure Report.
It requiers either a WebdriverIO.Browser or Mocha.Test object to interact with the report.

It looks like this package no longer exists. Is your question still relevant?

> We want to put the command history to the report:

Looks like you just need to use reporter interface - https://github.com/gemini-testing/testplane/blob/master/src/reporters/base.js. Example of implementation flat reporter - https://github.com/gemini-testing/testplane/blob/master/src/reporters/flat.js. It means you can get `test.history` inside methods `_onTestPass`, `_onTestFail` and as result save report on `_onRunnerEnd`.
